src/Elasticsearch/IndexBuilder.php

```diff
@@ -78,7 +78,14 @@ public static function getFieldsTypes(ModelInterface $model) : array
                     break;
                 case Column::TYPE_VARCHAR:
                 case Column::TYPE_CHAR:
-                    $fields[$column->getName()] = isset($model->elasticSearchNotAnalyzed) && !$model->elasticSearchNotAnalyzed ? 'text' : 'keyword';
+
+                    $elasticSearchNotAnalyzed = true;
+                    if (isset($model->elasticSearchAnalyzedFields[$column->getName()])) {
+                        $elasticSearchNotAnalyzed = false;
+                    } elseif (isset($model->elasticSearchNotAnalyzed) && !isset($model->elasticSearchAnalyzedFields[$column->getName()])) {
+                        $elasticSearchNotAnalyzed = (bool) $model->elasticSearchNotAnalyzed;
+                    }
```

Comment:
We don't need to verify that the field doesn't exists in the else statement because if it falls on the else we already know that it doesn't exists.
```suggestion
                    if (isset($model->elasticSearchAnalyzedFields[$column->getName()])) {
                        $elasticSearchNotAnalyzed = false;
                    } else {
                        $elasticSearchNotAnalyzed = (bool) ($model->elasticSearchNotAnalyzed ?? true);
                    }
```

Reply:
Made the default to true since the only way it could be false is if it falls on the if.